src/go/k8s/pkg/resources/certmanager/type_helpers.go

```diff
@@ -199,48 +208,41 @@ func NewClusterCertificates(
 		internalFQDN: fqdn,
 		clusterFQDN:  clusterFQDN,
 		logger:       logger,
-
-		kafkaAPI:          tlsDisabledAPICertificates(),
-		schemaRegistryAPI: tlsDisabledAPICertificates(),
-		adminAPI:          tlsDisabledAPICertificates(),
-		pandaProxyAPI:     tlsDisabledAPICertificates(),
 	}
 	var err error
-	if kafkaListeners := kafkaAPIListeners(cluster); len(kafkaListeners) > 0 {
-		cc.kafkaAPI, err = cc.prepareAPI(ctx, kafkaAPI, RedpandaNodeCert, []string{OperatorClientCert, UserClientCert, AdminClientCert}, kafkaListeners, &keystoreSecret)
-		if err != nil {
-			return nil, fmt.Errorf("kafka api certificates %w", err)
-		}
+	kafkaListeners := kafkaAPIListeners(cluster)
+	cc.kafkaAPI, err = cc.prepareAPI(ctx, "kafka-api", kafkaAPI, RedpandaNodeCert, "", []string{OperatorClientCert, UserClientCert, AdminClientCert}, kafkaListeners, &keystoreSecret)
+	if err != nil {
+		return nil, fmt.Errorf("kafka api certificates %w", err)
 	}
 
-	if adminListeners := adminAPIListeners(cluster); len(adminListeners) > 0 {
-		cc.adminAPI, err = cc.prepareAPI(ctx, adminAPI, adminAPINodeCert, []string{adminAPIClientCert}, adminListeners, &keystoreSecret)
-		if err != nil {
-			return nil, fmt.Errorf("kafka api certificates %w", err)
-		}
+	adminListeners := adminAPIListeners(cluster)
+	cc.adminAPI, err = cc.prepareAPI(ctx, "admin-api", adminAPI, adminAPINodeCert, "", []string{adminAPIClientCert}, adminListeners, &keystoreSecret)
+	if err != nil {
+		return nil, fmt.Errorf("admin api certificates %w", err)
 	}
 
-	if pandaProxyListeners := pandaProxyAPIListeners(cluster); len(pandaProxyListeners) > 0 {
-		cc.pandaProxyAPI, err = cc.prepareAPI(ctx, pandaproxyAPI, pandaproxyAPINodeCert, []string{pandaproxyAPIClientCert}, pandaProxyListeners, &keystoreSecret)
-		if err != nil {
-			return nil, fmt.Errorf("kafka api certificates %w", err)
-		}
+	pandaProxyListeners := pandaProxyAPIListeners(cluster)
```

Comment:
same as above

Reply:
Seems that panda proxy listener can be optional. I will check.